### PR TITLE
Failing AST test with non-named column

### PR DESCRIPTION
**Pull request by jakubvojacek**
Hello @staabm 

this is related to https://github.com/staabm/phpstan-dba/issues/566#issuecomment-1468512616, this is the reproducing case. When `count(*)` (or other functions) is used without alias, it fails to narrow down the type for integer indexed key in the resulting PDOStatement. 

I tried playing around with ParserInference but couldn't find a easy way to fix this yet

### CODE REVIEW

**Comment by staabm** [comment]
to implement this I think we need https://github.com/SQLFTW/sqlftw/commit/da4e0354e6d48dffb94a04a316b1b08b6a2ca38e which will be part of the next sqlftw release

**Comment by jakubvojacek** [comment]
perfect, cant wait for this

**Comment by staabm** [comment]
Today a new sqlftw was released. You can give it a try and update the lob with this PR and add a fix for you problem if you like

**Comment by jakubvojacek** [comment]
I will give it a go over the weekend! Thanks

**Comment by jakubvojacek** [comment]
The fix for the correct type inference was easy, but tests are failing @staabm 

We got some problems with postgres tests, since AST does not work there, it fails (https://github.com/staabm/phpstan-dba/actions/runs/4454918527/jobs/7824413966?pr=570)

For the same reason, it also fails for php 7.3 and lower

Any idea how to overcome this, please?

**Comment by staabm** [comment]
I think we should move 

https://github.com/staabm/phpstan-dba/blob/a1bb0c6e77f514528471e87d785084c21868b923/tests/default/DbaInferenceTest.php#L66-L68

into a separate test-class and enable the runtime-config for SQL AST only for this new test-class

**Comment by jakubvojacek** [comment]
Is this what you had in mind please?

File: tests/stringify/data/ast-narrowed-stringify.php [hunk 1]
```diff
@@ -13,6 +13,6 @@ public function count(PDO $pdo): void
         assertType('PDOStatement<array{myemail: numeric-string|null, 0: numeric-string|null}>', $stmt);
 
         $stmt = $pdo->query('SELECT count(email) as myemail, count(email) from ada');
-        assertType('PDOStatement<array{myemail: numeric-string|null, 0: numeric-string|null, count(email): numeric-string, 1: numeric-string|null}>', $stmt);
+        assertType('PDOStatement<array{myemail: numeric-string|null, 0: numeric-string|null, count(email): numeric-string|null, 1: numeric-string|null}>', $stmt);
```
Comment (staabm) [inline tests/stringify/data/ast-narrowed-stringify.php:16]: I think this test no longer runs with SQL AST enabled

**Review by staabm — CHANGES_REQUESTED** [review]
I don't think we need a whole new `defaultAst` suite, but just a new `SqlAstInferenceTest.php` within the `default` test-suite, which enables sql-ast like on setup/teardown, like we did for paramter validation in https://github.com/staabm/phpstan-dba/commit/bb2b8457f8a892f9ee59bc0fd1bdc285564c2292#diff-428d052928087e660593dfd032cc0445056ba9fe2780ddb2292983fefe04ac87R18-R26

**Pull request closed without merging**